src/js/components/DataTable/DataTable.js

```diff
@@ -152,7 +152,41 @@ const DataTable = ({
   const [scrollOffset, setScrollOffset] = useState(0);
 
   // multiple pinned columns offset
-  const [pinnedOffset, setPinnedOffset] = useState({});
+  const [pinnedOffset, setPinnedOffset] = useState();
+
+  // eslint-disable-next-line consistent-return
+  const getPinnedOffset = useCallback(() => {
+    const pinnedProperties = columns
+      .map(pinnedColumn => pinnedColumn.pin && pinnedColumn.property)
+      .filter(n => n);
+
+    const pinnedColumnsProperties = {};
+    const firstRowCells = headerRef.current.firstChild.cells || [];
+
+    if (firstRowCells.length !== 0) {
+      pinnedProperties.forEach((property, index) => {
+        const tableIndex = columns.findIndex(
+          column => column.property === property,
+        );
+
+        if (firstRowCells[tableIndex]) {
+          pinnedColumnsProperties[property] = {
+            width: firstRowCells[tableIndex].getBoundingClientRect()?.width,
+            left:
+              tableIndex === 0
+                ? 0
+                : pinnedColumnsProperties[pinnedProperties[index - 1]].left +
+                  pinnedColumnsProperties[pinnedProperties[index - 1]].width,
+          };
+        }
+      });
+      return pinnedColumnsProperties;
+    }
```

Comment:
```suggestion
    }
    return undefined;
```
Would it make sense to return undefined if we didn't set anything for `pinnedColumnsProperties` in this function? Effectively just reseting the state value to its default? If so, I think we could remove the eslint-disable comment